easybuild/easyconfigs/p/preCICE/preCICE-3.2.0-foss-2023a.eb

```diff
@@ -0,0 +1,64 @@
+easyblock = 'CMakeMake'
+
+name = 'preCICE'
+version = '3.2.0'
+
+homepage = 'https://www.precice.org/'
+description = """preCICE (Precise Code Interaction Coupling Environment)
+is a coupling library for partitioned multi-physics simulations,
+including, but not restricted to fluid-structure interaction and
+conjugate heat transfer simulations. Partitioned means that preCICE
+couples existing programs (solvers) capable of simulating a subpart of
+the complete physics involved in a simulation. This allows for the high
+flexibility that is needed to keep a decent time-to-solution for complex
+multi-physics scenarios."""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'usempi': True}
+
+source_urls = ['https://github.com/precice/precice/archive']
+sources = ['v%(version)s.tar.gz']
+checksums = ['93523f1a56e0cfd338d8e190baa06129ee811acdb1c697468a3c85c516d63464']
+
+builddependencies = [
+    ('CMake', '3.26.3'),
+    ('pkgconf', '1.9.5'),
+]
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('Boost', '1.82.0'),
+    ('Eigen', '3.4.0'),
+    ('libxml2', '2.11.4'),
+    ('PETSc', '3.20.3'),
+    ('SciPy-bundle', '2023.07'),
+]
+
+build_shared_libs = True
+
+# Tell pkg-config to keep flags available in CPATH and others because it prevents CMake from finding PETSc correctly
+preconfigopts = 'PKG_CONFIG_ALLOW_SYSTEM_LIBS=1 PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 '
+# Make sure MPI support is enabled
+configopts = '-DPRECICE_FEATURE_MPI_COMMUNICATION=ON '
+# Make sure PETSc support is enabled
+configopts += '-DPRECICE_FEATURE_PETSC_MAPPING=ON '
+# Make sure PythonActions support is enabled
+configopts += "-DPython3_ROOT_DIR=$EBROOTPYTHON "
```

Comment:
is this needed? The `CMakeMake` easyblock already sets:
```
-DPYTHON_EXECUTABLE=$EBROOTPYTHON/bin/python
-DPython_EXECUTABLE=$EBROOTPYTHON/bin/python
-DPython3_EXECUTABLE=$EBROOTPYTHON/bin/python 
```

Reply:
in fact, it also sets `Python3_ROOT_DIR` as an env var: https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/generic/cmakemake.py#L107

Reply:
42cd207

Reply:
This was written on EB4 and I'd forgotten to remove that.